tune/speed.h

```diff
@@ -3369,6 +3371,64 @@ int speed_routine_count_zeros_setup (struct speed_params *, mp_ptr, int, int);
   }
 
 
+/* Calculate worst case for perfect_power
+   Worst case is multiple prime factors larger than trial div limit.
+   Tests a number with approxiametly s->size bits. */
+#define SPEED_ROUTINE_MPN_PERFECT_POWER(function)   \
+  {									\
+    mpz_t     r, p;							\
+    unsigned  i, power;							\
+    double    t;							\
+									\
+    SPEED_RESTRICT_COND (s->size >= 1);					\
+									\
+    mpz_init(r);							\
+    mpz_init_set_ui (p, 20000);						\
+    mpz_nextprime(p, p);						\
```

Comment:
I could easily precalculate the next primes here and save two lines if that would be cleaner

Reply:
Took a great suggestion from Marco here